### PR TITLE
Adding filters for Table

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

- https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-317

I added the filters for the table, multiple and simple. ATM, it is missing sort by Date for the "Update" cell, I'll add this in a next merge request

Capture d’écran 2022-05-04 à 19 12 26

---

### Global

- [x] This PR does not introduce any breaking change
- [x] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

**Comment by RemiBonnet** [comment]
A preview environment was automatically created via Qovery.  
Click on the link below to follow its deployment and use it.  
:point_right: [[PR] staging - Table filter - 2022-05-03T17:18:39Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/845a6065-474a-43e3-8c8d-8ba927384007/applications)

**Comment by bdebon** [comment]
Hello @RemiBonnet, huge work here congratulations.
I'm testing the table filters and I have found another strange things than the one you mentioned (about the filter reopening when you click on the reset button)
The strange behavior I found here is shown in this video.

https://user-images.githubusercontent.com/6163954/166878607-52188a6e-2b1b-466c-9878-d9a518aaebad.mp4

Basically, when you have a filter on a column, and you click on another filter click the ALL button, it reset every filters from every rows when we would like to reset only the clicked column no?

File: libs/pages/environments/ui/src/lib/components/general/general.tsx [hunk 1]
```diff
@@ -12,31 +13,61 @@ export function GeneralPage(props: GeneralProps) {
...
+      defaultData={environments}
+      setFilterData={setData}
+      className="mt-2 bg-white rounded-sm"
+      columnsWidth="30% 15% 25% 10% 20%"
+    >
       <>
-        {environments.map((currentData, index) => (
+        {data.map((currentData, index) => (
           <TableRowEnvironments
             key={index}
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/pages/environments/ui/src/lib/components/general/general.tsx]: Yes of course, in some cases I can't do it but here it makes sense, thanks!

**Pull request merged**